plugins/VersionUpgrade/VersionUpgrade44to45/VersionUpgrade44to45.py

```diff
@@ -75,8 +94,42 @@ def upgradeStack(self, serialized: str, filename: str) -> Tuple[List[str], List[
         # Update version number.
         if "metadata" not in parser:
             parser["metadata"] = {}
+
+        # Check for leftover hidden machines. They should be deleted along with their associated files
+        stack_id = parser["general"]["id"]
+        if parser["metadata"]["setting_version"] == "10" and stack_id is not None:
+            stack_type = parser["metadata"]["type"]
+            if stack_type == "machine" and parser["metadata"]["hidden"]:
+                if not self._hidden_machines:
```

Comment:
If there are no hidden machines, the hidden machines need to be deleted? Shouldn't it be the other way around?